src/limitedmap.h

```diff
@@ -46,20 +46,22 @@ class limitedmap
         }
         return;
     }
-    void erase(const key_type& k)
+    size_t erase(const key_type& k)
     {
         iterator itTarget = map.find(k);
         if (itTarget == map.end())
-            return;
+            return 0;
         std::pair<rmap_iterator, rmap_iterator> itPair = rmap.equal_range(itTarget->second);
-        for (rmap_iterator it = itPair.first; it != itPair.second; ++it)
+        for (rmap_iterator it = itPair.first; it != itPair.second; ++it) {
             if (it->second == itTarget) {
                 rmap.erase(it);
                 map.erase(itTarget);
-                return;
+                return 1; // be consistent with std::map return value
             }
+        }
         // Shouldn't ever get here
         assert(0);
+        return 0;
```

Comment:
What's the point of this inserted `return 0` if, as the comment says plus the very strong assertion, we should never get here ?

Reply:
We need to set a standard : as of now I've been told we cannot disable assertions without breaking a whole lot more on the code so I would not see this as an issue.
On the other hand if you mean that we might have a warning for missing return on all code patterns then you might be right but this require either handling exceptions or return codes.

Reply:
Just for reference, there is an updated comment from Bitcoin (even though they removed the file years ago):

https://github.com/bitcoin/bitcoin/blob/cc16fff3e476a9378d2176b3c1b83ad12b1b052a/src/limitedmap.h#L69

Reply:
@AndreaLanfranchi @ptagl  Thank you guys for your comments. All in all, considering also the other comments, I agree with you that currently `src/limitedmap.h` has some problems and could use some refactoring and bugfixing.

I have some pending work already, and I will open another PR shortly dedicated to this, but in the meantime I would rather not pollute this PR and shift its original focus.

Is it ok with you if we go ahead with this merge as it is for now?

Reply:
I'm not on the same page but ok.